sdk/rust/oak_tests/src/lib.rs

```diff
@@ -938,39 +1000,43 @@ where
         .expect("failed to serialize GrpcRequest message");
 
     // Create a new channel for the response to arrive on and attach to the message.
-    let channel = RUNTIME.write().unwrap().new_channel();
+    let channel = RUNTIME.write().expect(RUNTIME_MISSING).new_channel();
     msg.channels
         .push(ChannelHalf::new(Direction::Write, channel.clone()));
     let read_half = ChannelHalf::new(Direction::Read, channel);
 
     // Send the message (with attached write handle) into the Node under test.
     let grpc_channel = RUNTIME
         .read()
-        .unwrap()
+        .expect(RUNTIME_MISSING)
         .grpc_channel()
         .expect("no gRPC channel setup");
-    grpc_channel.write().unwrap().write_message(msg);
+    grpc_channel
+        .write()
+        .expect("internal error: corrupt gRPC channel ref")
+        .write_message(msg);
 
     // Read the serialized, encapsulated response.
     loop {
         let mut size = 0u32;
         let mut count = 0u32;
-        let result = read_half.channel.write().unwrap().read_message(
-            std::usize::MAX,
-            &mut size,
-            std::u32::MAX,
-            &mut count,
-        );
-        if let Err(e) = result {
-            if e == OakStatus::OK.value() as u32 {
-                info!("no pending gRPC response message; poll again soon");
-                std::thread::sleep(std::time::Duration::from_millis(100));
-                continue;
-            } else {
-                panic!(format!("failed to read from response channel: {}", e));
+        let result = read_half
+            .channel
+            .write()
+            .expect("internal error: corrupt channel ref")
+            .read_message(std::usize::MAX, &mut size, std::u32::MAX, &mut count);
+        let rsp = match result {
```

Comment:
Could this be a continuation of the previous expression, with `unwrap_or_else` to handle the error case?

Reply:
(Deferring for now, if only to make my merges easier&hellip;)